### PR TITLE
Fix 4292

### LINKED ISSUE #4292: @authorization bug when querying nested connection objects
### Type definitions

```GraphQL
type User
{
  id: ID!
    @unique

  email: String!
    @unique

  name: String

  creator: [Group!]!
    @relationship(type: "CREATOR_OF", direction: OUT)

  admin: [Admin!]!
    @relationship(type: "IS_USER", direction: IN)

  contributor: [Contributor!]!
    @relationship(type: "IS_USER", direction: IN)

  invitations: [Invitee!]!
    @relationship(type: "CREATOR_OF", direction: OUT)

  roles: [String!]!
}

type Group
{
  id: ID!
    @id
    @unique

  name: String

  members: [Person!]!
    @relationship(type: "MEMBER_OF", direction: IN)

  creator: User!
    @relationship(type: "CREATOR_OF", direction: IN)
    @settable(onCreate: true, onUpdate: true)

  admins: [Admin!]!
    @relationship(type: "ADMIN_OF", direction: IN)

  contributors: [Contributor!]!
    @relationship(type: "CONTRIBUTOR_TO", direction: IN)

}

type Person
  @authorization(
    validate: [{
      operations: [CREATE],
      where: { node: { group: { creator: { roles_INCLUDES: "plan:paid" }}}},
    }, {
      operations: [DELETE],
      where: { OR: [
        { node: { creator: { id: "$jwt.uid" }}},
        { node: { group: { admins_SOME: { user: { id: "$jwt.uid" }}}}},
        { node: { group: { creator: { id: "$jwt.uid" }}}},
      ]}
    }, {
      operations: [READ, UPDATE],
      where: { OR: [
        { node: { creator: { id: "$jwt.uid" }}},
        { node: { group: { admins_SOME: { user: { id: "$jwt.uid" }}}}},
        { node: { group: { contributors_SOME: { user: { id: "$jwt.uid" }}}}},
        { node: { group: { creator: { id: "$jwt.uid" }}}},
      ]}
    }]
  )
{
  id: ID!
    @id
    @unique

  name: String!

  creator: User!
    @relationship(type: "CREATOR_OF", direction: IN, nestedOperations: [CONNECT])
    @settable(onCreate: true, onUpdate: true)

  group: Group!
    @relationship(type: "MEMBER_OF", direction: OUT)

  partners: [Person!]!
    @relationship(type: "PARTNER_OF", queryDirection: UNDIRECTED_ONLY, direction: OUT, properties: "PartnerOf")
}

enum InviteeRole {
  ADMIN
  CONTRIBUTOR
}

enum InviteeStatus {
  INVITED
  ACCEPTED
}

interface Invitee
{
  id: ID!
    @id

  email: String!

  name: String

  creator: User!
    @relationship(type: "CREATOR_OF", direction: IN)

  group: Group!
    @relationship(type: "ADMIN_OF", direction: OUT)

  status: InviteeStatus!
    @default(value: INVITED)

  user: User
    @relationship(type: "IS_USER", direction: OUT)

  role: InviteeRole!
}

type Admin implements Invitee
{
  id: ID!
    @unique
  group: Group!
  creator: User!
  email: String!
  name: String
  status: InviteeStatus!
  user: User
  role: InviteeRole!
    @default(value: ADMIN)
}

type Contributor implements Invitee {
  id: ID!
    @unique
  group: Group!
    @relationship(type: "CONTRIBUTOR_TO", direction: OUT)
  creator: User!
  email: String!
  name: String
  status: InviteeStatus!
  user: User
  role: InviteeRole!
    @default(value: CONTRIBUTOR)
}

interface PartnerOf @relationshipProperties {
  id: ID!
    @id
  firstDay: Date
  lastDay: Date
  active: Boolean!
    @default(value: true)
}

type JWT @jwt {
  roles: [String!]!
}

type Mutation {
  sendInvite(id: ID!, role: InviteeRole!): Boolean!
}

extend schema @authentication
```

### Test data

_No response_

### Steps to reproduce

Execute the following query:
```gql
query Groups($where: GroupWhere) {
  groups(where: $where) {
    id
    name
    members {
      id
      name
      partnersConnection {
        edges {
          active
          firstDay
          lastDay
        }
      }
    }
  }
}
```
with variables:
```json
{
  "where": {
    "id": "family_id_1"
  }
}
```

### What happened

```json
{
  "data": {},
  "errors": [
    {
      "message": "Variable `creatorCount` not defined (line 43, column 78 (offset: 2285))\n\"        WHERE apoc.util.validatePredicate(NOT ($isAuthenticated = true AND ((creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this14.id = NULL.uid)) OR (groupCount <> 0 AND size([(this15)<-[:ADMIN_OF]-(this22:Admin) WHERE single(this21 IN [(this22)-[:IS_USER]->(this21:User) WHERE (NULL.uid IS NOT NULL AND this21.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND size([(this16)<-[:CONTRIBUTOR_TO]-(this24:Contributor) WHERE single(this23 IN [(this24)-[:IS_USER]->(this23:User) WHERE (NULL.uid IS NOT NULL AND this23.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND (creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this25.id = NULL.uid))))), \"@neo4j/graphql/FORBIDDEN\", [0])\"\n                                                                              ^",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "groups"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "stacktrace": [
          "Neo4jError: Variable `creatorCount` not defined (line 43, column 78 (offset: 2285))",
          "\"        WHERE apoc.util.validatePredicate(NOT ($isAuthenticated = true AND ((creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this14.id = NULL.uid)) OR (groupCount <> 0 AND size([(this15)<-[:ADMIN_OF]-(this22:Admin) WHERE single(this21 IN [(this22)-[:IS_USER]->(this21:User) WHERE (NULL.uid IS NOT NULL AND this21.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND size([(this16)<-[:CONTRIBUTOR_TO]-(this24:Contributor) WHERE single(this23 IN [(this24)-[:IS_USER]->(this23:User) WHERE (NULL.uid IS NOT NULL AND this23.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND (creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this25.id = NULL.uid))))), \"@neo4j/graphql/FORBIDDEN\", [0])\"",
          "                                                  
```

### Expected behaviour

Expect the query to execute without error.

### Version

4.4.1

### Database version

5.12

### Relevant log output

```Shell
@neo4j/graphql:execution executing cypher +23ms
  @neo4j/graphql:execution MATCH (this:Group)
  @neo4j/graphql:execution WHERE this.id = $param0
  @neo4j/graphql:execution CALL {
  @neo4j/graphql:execution     WITH this
  @neo4j/graphql:execution     MATCH (this)<-[this0:MEMBER_OF]-(this1:Person)
  @neo4j/graphql:execution     OPTIONAL MATCH (this1)<-[:CREATOR_OF]-(this2:User)
  @neo4j/graphql:execution     WITH *, count(this2) AS creatorCount
  @neo4j/graphql:execution     OPTIONAL MATCH (this1)-[:MEMBER_OF]->(this3:Group)
  @neo4j/graphql:execution     WITH *, count(this3) AS groupCount
  @neo4j/graphql:execution     OPTIONAL MATCH (this1)-[:MEMBER_OF]->(this4:Group)
  @neo4j/graphql:execution     WITH *, count(this4) AS groupCount
  @neo4j/graphql:execution     WITH *
  @neo4j/graphql:execution     CALL {
  @neo4j/graphql:execution         WITH this1
  @neo4j/graphql:execution         MATCH (this1)-[:MEMBER_OF]->(this5:Group)
  @neo4j/graphql:execution         OPTIONAL MATCH (this5)<-[:CREATOR_OF]-(this6:User)
  @neo4j/graphql:execution         WITH *, count(this6) AS creatorCount
  @neo4j/graphql:execution         WITH *
  @neo4j/graphql:execution         WHERE (creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this6.id = NULL.uid))
  @neo4j/graphql:execution         RETURN count(this5) = 1 AS var7
  @neo4j/graphql:execution     }
  @neo4j/graphql:execution     WITH *
  @neo4j/graphql:execution     WHERE apoc.util.validatePredicate(NOT ($isAuthenticated = true AND ((creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this2.id = NULL.uid)) OR (groupCount <> 0 AND size([(this3)<-[:ADMIN_OF]-(this9:Admin) WHERE single(this8 IN [(this9)-[:IS_USER]->(this8:User) WHERE (NULL.uid IS NOT NULL AND this8.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND size([(this4)<-[:CONTRIBUTOR_TO]-(this11:Contributor) WHERE single(this10 IN [(this11)-[:IS_USER]->(this10:User) WHERE (NULL.uid IS NOT NULL AND this10.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR var7 = true)), "@neo4j/graphql/FORBIDDEN", [0])
  @neo4j/graphql:execution     CALL {
  @neo4j/graphql:execution         WITH this1
  @neo4j/graphql:execution         MATCH (this1)-[this12:PARTNER_OF]-(this13:Person)
  @neo4j/graphql:execution         OPTIONAL MATCH (this13)<-[:CREATOR_OF]-(this14:User)
  @neo4j/graphql:execution         WITH *, count(this14) AS creatorCount
  @neo4j/graphql:execution         OPTIONAL MATCH (this13)-[:MEMBER_OF]->(this15:Group)
  @neo4j/graphql:execution         WITH *, count(this15) AS groupCount
  @neo4j/graphql:execution         OPTIONAL MATCH (this13)-[:MEMBER_OF]->(this16:Group)
  @neo4j/graphql:execution         WITH *, count(this16) AS groupCount
  @neo4j/graphql:execution         OPTIONAL MATCH (this13)-[:MEMBER_OF]->(this17:Group)
  @neo4j/graphql:execution         WITH *, count(this17) AS groupCount
  @neo4j/graphql:execution         WITH *
  @neo4j/graphql:execution         MATCH (this13)-[:MEMBER_OF]->(this18:Group)
  @neo4j/graphql:execution         OPTIONAL MATCH (this18)<-[:CREATOR_OF]-(this19:User)
  @neo4j/graphql:execution         WITH *, count(this19) AS creatorCount
  @neo4j/graphql:execution         WITH *
  @neo4j/graphql:execution         WHERE (creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this19.id = NULL.uid))
  @neo4j/graphql:execution         RETURN count(this18) = 1 AS var20
  @neo4j/graphql:execution         WITH *
  @neo4j/graphql:execution         WHERE apoc.util.validatePredicate(NOT ($isAuthenticated = true AND ((creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this14.id = NULL.uid)) OR (groupCount <> 0 AND size([(this15)<-[:ADMIN_OF]-(this22:Admin) WHERE single(this21 IN [(this22)-[:IS_USER]->(this21:User) WHERE (NULL.uid IS NOT NULL AND this21.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND size([(this16)<-[:CONTRIBUTOR_TO]-(this24:Contributor) WHERE single(this23 IN [(this24)-[:IS_USER]->(this23:User) WHERE (NULL.uid IS NOT NULL AND this23.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND (creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this25.id = NULL.uid))))), "@neo4j/graphql/FORBIDDEN", [0])
  @neo4j/graphql:execution         WITH { active: this12.active, firstDay: this12.firstDay, lastDay: this12.lastDay, node: { __resolveType: "Person", __id: id(this13) } } AS edge
  @neo4j/graphql:execution         WITH collect(edge) AS edges
  @neo4j/graphql:execution         WITH edges, size(edges) AS totalCount
  @neo4j/graphql:execution         RETURN { edges: edges, totalCount: totalCount } AS var26
  @neo4j/graphql:execution     }
  @neo4j/graphql:execution     WITH this1 { .id, .name, partnersConnection: var26 } AS this1
  @neo4j/graphql:execution     RETURN collect(this1) AS var27
  @neo4j/graphql:execution }
  @neo4j/graphql:execution RETURN this { .id, .name, members: var27 } AS this +0ms
  @neo4j/graphql:execution cypher params: { param0: 'family_id_1', jwt: null, isAuthenticated: true } +1ms
  @neo4j/graphql:execution Neo4jError: Variable `creatorCount` not defined (line 43, column 78 (offset: 2285))
"        WHERE apoc.util.validatePredicate(NOT ($isAuthenticated = true AND ((creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this14.id = NULL.uid)) OR (groupCount <> 0 AND size([(this15)<-[:ADMIN_OF]-(this22:Admin) WHERE single(this21 IN [(this22)-[:IS_USER]->(this21:User) WHERE (NULL.uid IS NOT NULL AND this21.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND size([(this16)<-[:CONTRIBUTOR_TO]-(this24:Contributor) WHERE single(this23 IN [(this24)-[:IS_USER]->(this23:User) WHERE (NULL.uid IS NOT NULL AND this23.id = NULL.uid) | 1] WHERE true) | 1]) > 0) OR (groupCount <> 0 AND (creatorCount <> 0 AND (NULL.uid IS NOT NULL AND this25.id = NULL.uid))))), "@neo4j/graphql/FORBIDDEN", [0])"
```

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

Fix for issue #4292

## Complexity

Complexity: Low

Closes #4292 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4309.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 37 | 37 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 40 | 56 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 52 | 84 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 37 | 50 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 2189 | 2382 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2292 | 2299 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 33 | 40 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 28 | 42 | 2174 |
|  aggregations.DeeplyNestedCount | 11062443 | 11062443 | 3017 | 3221 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 38 | 48 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 74 | 78 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 126 | 154 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 46 | 75 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 150 | 213 | 3003 |
|  connections.Connection | 14082 | 14082 | 55 | 71 | 2174 |
|  connections.NestedConnection | 41118 | 41118 | 78 | 105 | 4516 |
|  create.SimpleMutation | 7 | 7 | 47 | 63 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 21 | 22 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 217 | 204 | 1040 |
|  delete.NestedDeleteInUpdate | 19198 | 19198 | 185 | 173 | 1183 |
|  2871.NestedRelationshipFilter | 23160 | 23142 | 94 | 67 | 4442 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 36 | 48 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22887 | 22900 | 70 | 90 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 35 | 43 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 47 | 58 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 19 | 21 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 35 | 40 | 2174 |
|  query.QueryWhere | 9715 | 9705 | 30 | 38 | 2167 |
|  query.SimpleQueryWithNestedWhere | 9893 | 9883 | 43 | 52 | 2167 |
|  query.Nested | 10096041 | 10096041 | 6108 | 6877 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 6037 | 6719 | 2004000 |
|  query.OrFilterOnRelationships | 42767 | 41489 | 132 | 155 | 1881 |
|  query.OrFilterOnRelationshipsAndNested | 37116 | 34366 | 180 | 225 | 1881 |
|  query.QueryWithNestedIn | 13564 | 13273 | 49 | 56 | 1269 |
|  query.NestedConnectionWhere | 9834 | 9834 | 46 | 57 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10084 | 9993 | 62 | 78 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19314 | 19242 | 134 | 140 | 1652 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 134 | 175 | 1134 |
|  query.Fulltext | 80 | 80 | 31 | 31 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 36 | 42 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14092 | 14092 | 43 | 49 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14092 | 14092 | 52 | 59 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14259 | 14189 | 59 | 70 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14259 | 14189 | 73 | 87 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14615 | 14549 | 93 | 118 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14543 | 14473 | 84 | 109 | 2174 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 53 | 63 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14487 | 14584 | 76 | 85 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 44 | 46 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 71 | 92 | 4516 |
|  sorting.ConnectionWithSort | 3284 | 3284 | 66 | 53 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 58 | 55 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 51 | 62 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 255 | 265 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 250 | 244 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 87 | 109 | 2002 |

</details>
        
Old Schema Generation: 24.876s
Schema Generation: 24.481s
Old Subgraph Schema Generation: 28.573s
Subgraph Schema Generation: 28.528s

File: packages/graphql/tests/integration/issues/4292.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    const User = new UniqueType("User");
+    const Group = new UniqueType("Group");
+    const Person = new UniqueType("Person");
+    const Admin = new UniqueType("Admin");
+    const Contributor = new UniqueType("Contributor");
+
+    beforeAll(async () => {
+        neo4j = new Neo4j();
+        driver = await neo4j.getDriver();
+        const typeDefs = gql`
```
Comment (mjfwebb) [inline packages/graphql/tests/integration/issues/4292.int.test.ts:41]:
```suggestion
        const typeDefs = /* GraphQL */`
```

Comment (mjfwebb) [reply packages/graphql/tests/integration/issues/4292.int.test.ts:41]: The same for the other file, and to remove the imports :)

**Review by mjfwebb — APPROVED** [review]
Thank you!

**Pull request merged**